connectivity/netsocket/include/netsocket/MsgHeader.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2021 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MSG_HEADER_H
+#define MSG_HEADER_H
+
+#include "netsocket/nsapi_types.h"
+
+/**
+ * Allows iteration through the list of message headers received in the control parameter of the
```

Comment:
We need doxygen that demonstrate how this can be used as well as individual doxygen comment for public functions.

Reply:
Where would you like me to add doxygen with usage example, somewhere in the new file ?

Reply:
No you can add it to the class documentation. Here's and example for the Span class: https://github.com/ARMmbed/mbed-os/blob/c14396a5b619b185d344ad6dfc25f04cb330f1fc/platform/include/platform/Span.h#L74-L213

The code goes between the `@code` and `@endcode` blocks.

Reply:
Done, thanks